src/l500/l500-device.cpp

```diff
@@ -710,47 +709,40 @@ namespace librealsense
                         }
                
                         std::lock_guard<std::mutex> lock(_temperature_mutex);
-                        auto t = reinterpret_cast<temperatures*>((void*)res.data());
-                        _temperatures->LDD_temperature = t->LDD_temperature;
-                        _temperatures->MC_temperature = t->MC_temperature;
-                        _temperatures->MA_temperature = t->MA_temperature;
-                        _temperatures->APD_temperature = t->APD_temperature;
-                        _temperatures->HUM_temperature = t->HUM_temperature;
-                        _temperatures->AlgoTermalLddAvg_temperature = t->AlgoTermalLddAvg_temperature;
-               
-                        if (fw_version_support_nest)
-                        {
-                            extended_temperatures *t_ex = reinterpret_cast<extended_temperatures*>((void*)res.data());
-                            _temperatures->nest_avg = t_ex->nest_avg;
-                        }
 
-                        _temperature_from_fetcher = true;
+                        memset(&_temperatures, sizeof(_temperatures), 0);
+                        if (fw_version_support_nest)
+                            _temperatures = *reinterpret_cast<extended_temperatures const *>(res.data());
+                        else
+                            *reinterpret_cast<temperatures *>(&_temperatures) = *reinterpret_cast<temperatures const *>(res.data());
+                       
+                        _have_temperatures = true;
                     }
                
-                    // Do not hold the thread alive too long if fetcher were turned off
-                    std::this_thread::sleep_for(std::chrono::milliseconds(50));
+                    // Do not hold the thread alive too long if reader threat was turned off
```

Comment:
thread

Reply:
Changed to // Do not hold the device alive too long if reader thread was turned off